modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -253,12 +271,115 @@ void GOCVBGR::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
             cv::util::throw_error(
                     std::logic_error("Unsupported MediaFormat for cv::gapi::streaming::BGR"));
     }
-    out.post(std::move(out_arg));
+}
+
+struct GOCVY: public cv::detail::KernelTag
+{
+    using API = cv::gapi::streaming::GY;
+    static cv::gapi::GBackend backend() { return cv::gapi::streaming::backend(); }
+
+    class Actor final: public GAccessorActorBase {
+    public:
+        using GAccessorActorBase::GAccessorActorBase;
+        virtual void extractRMat(const cv::MediaFrame& frame, cv::RMat& rmat) override;
+    };
+
+    static cv::gapi::streaming::IActor::Ptr create(const cv::GCompileArgs& args)
+    {
+        return cv::gapi::streaming::IActor::Ptr(new Actor(args));
+    }
+    static cv::gapi::streaming::GStreamingKernel kernel() { return {&create}; };
+};
+
+void GOCVY::Actor::extractRMat(const cv::MediaFrame& frame, cv::RMat& rmat)
+{
+    const auto& desc = frame.desc();
+    switch (desc.fmt)
+    {
+        case cv::MediaFormat::BGR:
+            {
+                auto view = frame.access(cv::MediaFrame::Access::R);
+                cv::Mat tmp_bgr(desc.size, CV_8UC3, view.ptr[0], view.stride[0]);
+                cv::Mat yuv;
+                cvtColor(tmp_bgr, yuv, cv::COLOR_BGR2YUV_I420);
```

Comment:
In case of `BGR()` accessor called on an NV12 MediaFrame, we simply throw an exception (except some target platform where that conversion is implemented in firmware).

Since in case of `Y()` accessor we actually do the conversion (on CPU - what is debatable for high resolutions we target to), I'd align the behavior between the accessors.

Also, since the on-the-fly conversion is costly, I'd put a warning but **only once** per actor - not for every frame we process. You might want to look at `std::call_once` for that.

Reply:
We don't throw exception in the OCV BGR kernel also..

Reply:
Added warning for every accessor in case of non-consistent input format.
Warning is thrown only once per frame.